### PR TITLE
Feat: request token when vela auth gen-kubeconfig

### LINKED ISSUE #4675: `vela auth gen-kubeconfig` incompatibility in K8s 1.24
**Describe the bug**

In k8s v1.24+. The SA's secret won't be generated automatically by default per https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.24.md#no-really-you-must-read-this-before-you-upgrade 

In `vela auth gen-kubeconfig --serviceaccount`, it read this secret and generate kubeconfig.

**To Reproduce**

run `vela auth gen-kubeconfig --serviceaccount` in K8s v1.24

**Expected behavior**

Kubeconfig can be generated.

**KubeVela Version**

main branch

**Cluster information**

**Additional context**

### PR DESCRIPTION
**Pull request by chivalryq**
Signed-off-by: Qiaozp <qiaozhongpei.qzp@alibaba-inc.com>
### Description of your changes

Fixes #4675 

With this PR, KubeVela is generally compatible with K8s 1.24.

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/auth/kubeconfig.go [hunk 1]
```diff
@@ -329,30 +335,53 @@ func generateX509KubeConfigV1Beta(ctx context.Context, cli kubernetes.Interface,
...
+				Audiences:         []string{},
+				ExpirationSeconds: pointer.Int64(int64(opts.ExpireTime.Seconds())),
+			},
+		}
+		tokenRequest, err := cli.CoreV1().ServiceAccounts(opts.ServiceAccountNamespace).CreateToken(ctx, opts.ServiceAccountName, &request, metav1.CreateOptions{})
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to request token")
+		}
+		token = tokenRequest.Status.Token
+		CAConfigMap, err := cli.CoreV1().ConfigMaps(sa.Namespace).Get(ctx, "kube-root-ca.crt", metav1.GetOptions{})
```
Comment (Somefive) [inline pkg/auth/kubeconfig.go:360]: lowercase CAConfigMap -> caConfigMap

File: pkg/cmd/builder.go [hunk 2]
```diff
@@ -116,7 +116,7 @@ func (option NamespaceFlagDisableEnvOption) ApplyToNamespaceFlagOptions(cfg *Nam
 // WithNamespaceFlag add namespace flag to the command, by default, it will also add env flag to the command
 func (builder *Builder) WithNamespaceFlag(options ...NamespaceFlagOption) *Builder {
 	cfg := newNamespaceFlagOptions(options...)
-	builder.cmd.Flags().StringP(flagNamespace, "n", "", cfg.usage)
+	builder.cmd.Flags().StringP(flagNamespace, "n", "default", cfg.usage)
```
Comment (Somefive) [inline pkg/cmd/builder.go:119]: Why set `default` here?

File: references/cli/auth.go [hunk 3]
```diff
@@ -153,7 +153,7 @@ func NewGenKubeConfigCommand(f velacmd.Factory, streams util.IOStreams) *cobra.C
 	}
 	cmd.Flags().StringVarP(&o.User, "user", "u", o.User, "The user of the generated kubeconfig. If set, an X509-based kubeconfig will be intended to create. It will be embedded as the Subject in the X509 certificate.")
 	cmd.Flags().StringSliceVarP(&o.Groups, "group", "g", o.Groups, "The groups of the generated kubeconfig. This flag only works when `--user` is set. It will be embedded as the Organization in the X509 certificate.")
-	cmd.Flags().StringVarP(&o.ServiceAccount, "serviceaccount", "", o.ServiceAccount, "The serviceaccount of the generated kubeconfig. If set, a kubeconfig will be generated based on the secret token of the serviceaccount. Cannot be set when `--user` presents.")
+	cmd.Flags().StringVarP(&o.ServiceAccount, "serviceaccount", "", "", "The serviceaccount of the generated kubeconfig. If set, a kubeconfig will be generated based on the secret token of the serviceaccount. Cannot be set when `--user` presents.")
```
Comment (Somefive) [inline references/cli/auth.go:156]: Any reason for change here?

Comment (chivalryq) [reply references/cli/auth.go:156]: I have no idea why we put "o.ServiceAccount" here. This arguments means default value. And default value should be empty.

File: pkg/cmd/builder.go (hunk 2 above)
Comment (chivalryq) [reply pkg/cmd/builder.go:119]: align with behavior of kubectl and other tools. No namespace means default.

File: pkg/auth/kubeconfig.go (hunk 1 above)
Comment (chivalryq) [reply pkg/auth/kubeconfig.go:360]: CA is abbreviation. Like HTTP. I thought it should be uppercase. 🤔

Comment (Somefive) [reply pkg/auth/kubeconfig.go:360]: The upper case usually indicates public access to variables. I suggest using caConfigMap, but well, it is okay to work with CAConfigMap.

File: pkg/cmd/builder.go (hunk 2 above)
Comment (Somefive) [reply pkg/cmd/builder.go:119]: I think no explicit namespace flag will first resolve the namespace from the local config? Like you can set namespace value in `.kube/config` and kubectl will automatically switch to that.

File: references/cli/auth.go (hunk 3 above)
Comment (Somefive) [reply references/cli/auth.go:156]: Yes, the default value is empty, and so does o.User & o.Groups. I suggest aligning all the option style here. If you would like to change o.ServiceAccount here to empty, then o.User & o.Groups should be changed as well (but still I suggest not changing here.)

File: pkg/auth/kubeconfig.go (hunk 1 above)
Comment (chivalryq) [reply pkg/auth/kubeconfig.go:360]: 👌 Let's keep it. After all it's a local variable. No need to worry about public access.

File: pkg/cmd/builder.go (hunk 2 above)
Comment (chivalryq) [reply pkg/cmd/builder.go:119]: Oh I didn't considered that. Thanks.

File: references/cli/auth.go (hunk 3 above)
Comment (chivalryq) [reply references/cli/auth.go]: Will rollback

File: references/cli/auth.go [hunk 4]
```diff
@@ -153,7 +153,7 @@ func NewGenKubeConfigCommand(f velacmd.Factory, streams util.IOStreams) *cobra.C
 	}
 	cmd.Flags().StringVarP(&o.User, "user", "u", o.User, "The user of the generated kubeconfig. If set, an X509-based kubeconfig will be intended to create. It will be embedded as the Subject in the X509 certificate.")
 	cmd.Flags().StringSliceVarP(&o.Groups, "group", "g", o.Groups, "The groups of the generated kubeconfig. This flag only works when `--user` is set. It will be embedded as the Organization in the X509 certificate.")
-	cmd.Flags().StringVarP(&o.ServiceAccount, "serviceaccount", "", o.ServiceAccount, "The serviceaccount of the generated kubeconfig. If set, a kubeconfig will be generated based on the secret token of the serviceaccount. Cannot be set when `--user` presents.")
+	cmd.Flags().StringVarP(&o.ServiceAccount, "serviceaccount", o.ServiceAccount, "", "The serviceaccount of the generated kubeconfig. If set, a kubeconfig will be generated based on the secret token of the serviceaccount. Cannot be set when `--user` presents.")
```
Comment (wonderflow) [inline references/cli/auth.go:156]: the order changed?

Comment (chivalryq) [reply references/cli/auth.go]: Didn't mean that. A mistake when rollback.

**Comment by wonderflow** [comment]
do we backport to 1.5.x?

**Comment by chivalryq** [comment]
> do we backport to 1.5.x?

Yes, label added.

File: pkg/auth/kubeconfig.go (hunk 1 above)
Comment (Somefive) [reply pkg/auth/kubeconfig.go:360]: ok

File: references/cli/auth.go (hunk 3 above)
Comment (Somefive) [reply references/cli/auth.go]: ok

**Review by Somefive — APPROVED** [review]
Great Work!

**Pull request merged**